scripts/pages/zoning/zoning.ts

```diff
@@ -156,10 +159,12 @@ class ZoneMenuHandler {
 		this.teleDestList.unshift($.Localize('#Zoning_TPDest_None'));
 		this.populateDropdown(this.teleDestList, this.panels.regionTPDest, '', true);
 
+		this.zoningLimits = this.panels.zoningMenu.getZoningLimits();
+
 		this.createTrackEntry(this.panels.trackList, this.mapZoneData.tracks.main, $.Localize('#Zoning_Main'));
 
 		// if the first segment has no start zone, show text to create start zone until a start zone has been made
-		if (this.mapZoneData.tracks.main.zones.segments[0]?.checkpoints.length === 0) {
+		if (this.mapZoneData.tracks.main.zones?.segments?.length > 0 && this.mapZoneData.tracks.main.zones?.segments[0]?.checkpoints?.length) {
```

Comment:
Don't think you need this length check. `zones?.segments[0]?.` is null if array is length 0, which `?.`s carry to the end of the expression, so the final `.length` is either a number or `null`